### PR TITLE
10539 bokehjs.rst doc edit

### PR DESCRIPTION
**Pull request by Konntroll**
Finished editing bokehjs.rst for task #10539.

Please let me know if you would like me to make any changes to the text.

### CODE REVIEW

**Comment by mattpap** [comment]
For some reason you are accumulating changes from your previous PRs. There should be only commit 75013a6 present in this one. Seeing the `verified` tag on some of those commits, I presume you're using github's editor may have something to do with this. In any case, you will need to rebase all of your PRs and remove unrelated changes.

**Comment by Konntroll** [comment]
@mattpap, alright, will give it a go later. I was wondering about these change trails. I'm fairly new to git, so please bear with me.

**Comment by Konntroll** [comment]
@mattpap, I'm not sure if this is the way it's supposed to work. Looks like extraneous changes are now gone, though the IDs seem off, i.e. a00c9f7 to d6208af, rather than the original 75013a6. Am I doing it wrong?

**Comment by mattpap** [comment]
Every time you change (git) history, IDs change, as they are computed based on commits' contents and their predecessors. So that's fine, however, the single commit that there currently is, contains changes to three files. To fix that locally (I'm not sure if you can do that in github's UI), issue:
```sh
git reset branch-2.3 path_to_file_0
git reset branch-2.3 path_to_file_1
git commit --amend
git checkout .
git push -f
```
I would also make a backup branch (`git checkout -b tmp`) if you're not too comfortable with git. You can always `git reset --hard tmp` to restore your original work.

**Comment by mattpap** [comment]
> File contains trailing whitespace

Assuming you're editing locally, you can save yourself a bit of manual labor and our CI resources, by setting up your editor with automatic trailing whitespace removal, and running code quality checks locally (at minimum `pytest test/codebase/test_code_quality`).

**Comment by Konntroll** [comment]
Thanks for the pointers, @mattpap. I must have skipped some obvious step though as the extraneous files are still there. There's an option to delete them via GitHub UI if that's any good?

Ran ``pytest`` on this one as you suggested and it kept turning up trailing whitespace in layout.rst though there wasn't any on the lines it indicated. Switched on "Strip trailing white space on save" in PyCharm and it does strip the white space I add to the file but still won't fix the issues ``pytest`` is complaining of.

File: sphinx/source/docs/user_guide/bokehjs.rst [hunk 1]
```diff
@@ -3,72 +3,70 @@
...
-Low-Level Models
+Low-level models
 ----------------
 
-The low-level models for building up plots and applications (e.g. guides
-and glyphs and widgets, etc.) generally match the Bokeh Python models
-exactly. Accordingly, the :ref:`refguide` is a primary resource for
-answering questions about BokehJS models, even though it is presented
-from a Python perspective.
+Generally, the low-level models for plots and applications (e.g. guides,
```
Comment (tcmetzger) [inline sphinx/source/docs/user_guide/bokehjs.rst:32]: To improve legibility, Google's style guide encourages using something like "for example" instead of the Latin "e.g." (https://developers.google.com/style/abbreviations?hl=vi#dont-use)

**Comment by bryevdv** [comment]
@Konntroll Thanks for your patience, there's definitely no rush and your edits are very good ones, so we can take time and get everything right, and we are happy to help. The current issue with the PR is that commit 820f58c actually deletes the file itself, rather than undoing the changes to the file (i.e. if this PR were merged now, that file would be deleted from the repository). So let's start by reverting that commit. Executing this should restore the file: 
```
git revert 820f58c
``` 
If you do that and push then if there still changes showing up in the PR, we can next figure out how to revert those changes (without deleting fike)

**Comment by bryevdv** [comment]
As for the whitespace, I am not sure about your editor. But one way I have found to see whitespace in the PR on GitHub is to highlight the text, and they the trailing whitespace is definitely visible:

Screen Shot 2020-10-14 at 9 41 01 AM

**Comment by Konntroll** [comment]
Thanks for you help, @bryevdv. I'll try fiddling around with these PRs some more tonight.

I'm done with graph.rst but the PR still looks like it would be dragging along deleted jupyter.rst and a host of other files that never showed in staging, so I'll have to wait on creating that PR until I figure out how to fix the mess I've made of the repo.

**Comment by tcmetzger** [comment]
@Konntroll When I started with GitHub, I found [GitHub desktop](https://desktop.github.com/) to be really helpful - I felt like it gave me a better idea of what is actually happening and what will be included in a push. Maybe give that a try? 
Like @bryevdv has said previously, I would recommend just abandoning these current pull requests (we'll just close them, no problem) and creating new pull requests for each of the files. To make sure there are no further mixups, I'd recommend [creating a new branch](https://docs.github.com/en/free-pro-team@latest/desktop/contributing-and-collaborating-using-github-desktop/managing-branches#creating-a-branch) for _each_ file you are working on (in your [personal fork](https://docs.github.com/en/free-pro-team@latest/desktop/contributing-and-collaborating-using-github-desktop/cloning-and-forking-repositories-from-github-desktop#forking-repositories) of the Bokeh repo). Then make sure your commits each [only contain the _one_ file you want to include](https://docs.github.com/en/free-pro-team@latest/desktop/contributing-and-collaborating-using-github-desktop/committing-and-reviewing-changes-to-your-project#2-selecting-changes-to-include-in-a-commit).
Regarding the whitespaces: I also had some issues with those in the beginning - I'm using VS Code and I added the extension "[Trailing Spaces](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces)" to highlight and delete whitespaces. I'm sure you can find something similar for your IDE!

**Comment by Konntroll** [comment]
Thank you for all the advice, help, and patience, @tcmetzger, @mattpap, @bryevdv! I'll try to take it all on board as I go but please excuse the unavoidable hiccups yet to come :-)

Think I managed to create a cleanish PR for graph.rst. Saying that, I forgot to trim whitespace from it... :-|

**Comment by bryevdv** [comment]
@Konntroll that you for the work on the docs updates! I think the main thing (which is not obvious to start) is that when you create new git braches, they are *relative* to whatever branch you started on, so e.g if you do something like. 
```
git checkout branch-2.3

git checkout -b feature-1

<commit changes>

git checkout -b feature-2

```
then the the `feature-2` branch will have all the changes from `feature-1` as well, since that's was active when you created `feature-2`. I expect that is what happened here. 

TLDR always just check out the "base" branch first, before creating any branches for PRs. For Bokeh, base branches will always be like "branch-2.3" etc

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks @Konntroll !
